File: eunna/level3/43162.py
def solution(n, computers):
    answer = 0
    linked_computers = []
    for i in range(n):
        linked_computer = []
        for j in range(n):
            if i != j and computers[i][j] == 1:
                linked_computer.append(j)
        linked_computers.append(linked_computer)
    
    networks = []
    visited = [False] * n
    
    # 모든 컴퓨터의 방문 여부를 확인하기 위함
    for i in range(n):
        
        # 방문한 컴퓨터인지 확인
        if visited[i]:
            continue

        stack = [i]
        network = []

        while stack:
            computer = stack.pop()
            if computer not in network:
                network.append(computer)
                stack += linked_computers[computer]
                visited[computer] = True

        networks.append(network)
                    
    answer = len(networks)
    
    return answer

File: seon/level3/43162.js
function dfs(node, visit, computers) {
  const stack = [node];

  while (stack.length > 0) {
    const curNode = stack.pop();
    if (visit[curNode]) continue;
    let nextNode = 0;
    const lastNode = computers[curNode].length - 1;

    visit[curNode] = true;
    while (nextNode <= lastNode) {
      if (computers[curNode][nextNode]) stack.push(nextNode);
      nextNode += 1;
    }
  }
}

function solution(n, computers) {
  const visit = [];
  visit.length = n;
  visit.fill(false);

  let answer = 0;
  for (let node = 0; node < visit.length; node++) {
    if (!visit[node]) {
      dfs(node, visit, computers);
      answer += 1;
    }
  }

  return answer;
}
